source/concepts/protocols.rst

```diff
@@ -0,0 +1,310 @@
+```````````
+
+In this sub-phase, the protocol tries to create an agreement on the state to be
+settled without any on-chain transactions. If it succeeds, then the channel
+direclty progresses to the ``settle phase``. If not, then it transitions to the
+next sub-phase. This protocol is same for all three types of channels.
+
+If the latest off-chain state was marked as ``final`` already in the transact
+phase, it implies both the participants have agreed it to be the state to be
+settled. If it was not, another another off-chain transaction is sent, on behalf
+of the participant who initiated the channel settlement, marking the latest
+off-chain state as ``final``. It is up to the other participant to accept or
+reject or ignore it.
+
+If accepted, channel transitions to ``settle phase``, if not it transitions to
+the next sub-phase.
+
+.. image:: ../_generated/concepts/register_generic_finalize.svg
+  :align: Center
+  :alt: Image not available
+
+
+2. Register dispute on-chain
+````````````````````````````
+
+In this sub-phase, the protocol creates an agreement on the state to be
+settled by registering a dispute with the parent layer and resolving it.
+
+The parent layer for registering dispute is always the blockchain. In case of
+
+1. Ledger channel: The state of the channel, all its sub-channels and virtual
+   channels must be collected and registered on the blockchain.
+
+2. Sub-channel: The state of the parent channel, all the sub-channels and
+   virtual channels of the parent channel must be collected and registered on
+   the blockchain.
+
+3. Virtual channel: The state of the parent channel between the participant who
+   initiated channel settlement and the intermediary, all the sub-channels and
+   virtual channels of this parent channel must be collected and registered on
+   the blockchain.
+
+   Once the common intermediary is notified that one of the parent channels
+   holding funds for the virtual channel has been registered on-chain for
+   dispute resolution, the intermediary will try to finalize the state of the
+   virtual channel with the other channel participant (who did not initiate
+   settlement) through an off-chain transaction. If is succeeds, then that the
+   funds locked in the ledger channel between the intermediary and the other
+   participant will be settled off-chain. If not, then this ledger channel,
+   along with all its sub-channels and virtual channels will be registered on
+   the blockchain for dispute resolution.
+
+
+.. image:: ../_generated/concepts/register_generic_register.svg
+  :align: Center
+  :alt: Image not available
+
+After the challenge duration for register expires, if the channel has
+
+    1. ``No app``, then this state can be settled.
+    2. ``An app``, the channel transitions to next sub-phase.
+
+3. Progress the channel state on the blockhain
+``````````````````````````````````````````````
+
+This is a special sub-phase of register, relevant only for channels that have an
+app.
```

Comment:
Where is the app concept described? Is it possible to add a link for further information? Otherwise it is hard to get it, as it is not mentioned before.